### PR TITLE
[flutter_tools] remove some globals from flutter_tester device

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove globals from flutter_tester device and cleanup test case. Not completely gone since the Kernel Builder will still use them, but a good incremental improvement.

### CODE REVIEW

File: packages/flutter_tools/lib/src/tester/flutter_tester.dart [hunk 1]
```diff
@@ -38,17 +39,42 @@ class FlutterTesterApp extends ApplicationPackage {
...
+  final ProcessManager _processManager;
+  final FlutterVersion _flutterVersion;
+  final Logger _logger;
+  final String _buildDirectory;
+  final FileSystem _fileSystem;
+  final Artifacts _artifacts;
 
   Process _process;
-  final DevicePortForwarder _portForwarder = _NoopPortForwarder();
+  final DevicePortForwarder _portForwarder = const NoOpDevicePortForwarder();
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/tester/flutter_tester.dart:77]: Already had a no-op forwarder for desktop

File: packages/flutter_tools/lib/src/tester/flutter_tester.dart [hunk 2]
```diff
@@ -106,9 +132,6 @@ class FlutterTesterDevice extends Device {
   @override
   bool isSupported() => true;
 
-  bool _isRunning = false;
-  bool get isRunning => _isRunning;
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/tester/flutter_tester.dart:110]: I guess this was this unused?

**Review by jmagman — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/tester/flutter_tester.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/tester/flutter_tester.dart:110]: completely!

**Pull request merged**